tests/test_embedding_service.py

```diff
@@ -0,0 +1,111 @@
+import unittest
+import logging
+import tempfile
+import os
+import random
+import psutil
+from pathlib import Path
+
+from biotrainer.embedders import OneHotEncodingEmbedder, EmbeddingService
+from biotrainer.protocols import Protocol
+
+logger = logging.getLogger(__name__)
+
+class TestEmbeddingService(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._configure_logging()
+    
+    @classmethod
+    def _configure_logging(cls):
+        logging.basicConfig(
+            level=logging.DEBUG,
+            format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+            handlers=[logging.StreamHandler()]
+        )
+        logger.info('Setting up the test case class')
+
+    def setUp(self):
+        self._setup_test_environment()
+        self._setup_fasta_parameters()
+
+    def _setup_test_environment(self):
+        logger.info('Setting up test environment')
+        self.embedder = OneHotEncodingEmbedder()
+        self.embedding_service = EmbeddingService(embedder=self.embedder)
+
+    def _setup_fasta_parameters(self):
+        self.num_reads = 100
+        if os.environ.get('CI') == 'true':
```

Comment:
Nice solution, and seems to work as the tests took about 11mins. Did you double check that it works in the CI pipeline though? Might be nice to put a logging statement here to be able to make it easier understandable in the pipeline, why the test takes so long (for new developers for example).